src/item.cpp

```diff
@@ -6353,7 +6358,7 @@ std::string item::components_to_string() const
 
 bool item::needs_processing() const
 {
-    return active || has_flag( "RADIO_ACTIVATION" ) ||
+    return active || has_flag( "RADIO_ACTIVATION" ) || has_flag( "NANOFAB_TEMPLATE" ) ||
```

Comment:
This means items with this flag will *always* be processed on *every* turn.

Have you tried to set up the "NANOFAB_ITEM_ID" in the ìtem` constructor? Or a add a function `item::get_nanofab_item_id()` that checks whether the item needs such an id and either returns the existing id or creates a new one and returns it.